src/Backups/BackupImpl.cpp

```diff
@@ -387,6 +386,59 @@ void BackupImpl::readBackupMetadata()
     }
 }
 
+void BackupImpl::checkBackupDoesntExist() const
+{
+    String file_name_to_check_existence;
+    if (use_archives)
+        file_name_to_check_existence = archive_params.archive_name;
+    else
+        file_name_to_check_existence = ".backup";
+
+    if (writer->fileExists(file_name_to_check_existence))
+        throw Exception(ErrorCodes::BACKUP_ALREADY_EXISTS, "Backup {} already exists", backup_name);
+
+    /// Check that no other backup (excluding internal backups) is writing to the same destination.
+    if (!is_internal_backup)
+    {
+        assert(!lock_file_name.empty());
+        if (writer->fileExists(lock_file_name))
+            throw Exception(ErrorCodes::BACKUP_ALREADY_EXISTS, "Backup {} is being written already", backup_name);
+    }
+}
+
+void BackupImpl::createLockFile()
+{
+    /// Internal backup must not create the lock file (it should be created by the initiator).
+    assert(!is_internal_backup);
+
+    assert(uuid);
+    auto out = writer->writeFile(lock_file_name);
+    writeUUIDText(*uuid, *out);
+}
+
+bool BackupImpl::checkLockFile(bool throw_if_failed) const
+{
```

Comment:
@vitlibar , We are creating lockFile only when it is not internal backup .
However , checkLockFile(true) doesn't check if it is internal_backup or not and throws execution if it doesn't exist. 

Shouldn't we check here , if is_internal_backup we shouldn't check file existence of lock_file_name. 

please correct me if I am wrong.

Reply:
Because currently, I am facing issue When I am executing backup query on CLUSTER  , it throws error with ".lock suddenly disappeared while backup"
However query without ON cluster works fine . 

```
BACKUP TABLE tbl ON CLUSTER 'cluster' TO  Disk('backups', '02_9_2022') SETTINGS replica_num=1

Received exception from server (version 22.8.1):
Code: 655. DB::Exception: Received from localhost:9440. DB::Exception: Got error from <hostname>:9440. DB::Exception: Lock file .lock suddenly disappeared while writing backup Disk('backups', '02_9_2022'). (FAILED_TO_SYNC_BACKUP_OR_RESTORE)
```

Reply:
That's right the lock file is created only if it's not internal backup but checked for existence no matter if it's internal backup or not. How it works:

Reply:
Thanks @vitlibar for info. 
I see createLockFile() is called only once here https://github.com/ClickHouse/ClickHouse/blob/master/src/Backups/BackupImpl.cpp#L212
which forced me to think `is_internal_backup `is true only for worker nodes and false for initiator node.
please correct me if I am wrong. 


And I would like to know your suggestions on the problem I ma facing with backup ON Cluster command in  my environment.